File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashTable = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if nums[i] in hashTable:
                return [i, hashTable.get(nums[i])]
            else:
                hashTable[complement] = i
            

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashMap = new Map();
    
    for (var i = 0; i < nums.length; i++){
        var complement = target - nums[i]
        if (!hashMap.get(nums[i])) {
            hashMap.set(complement, i);
        }
    }
    for (var i = 0; i < nums.length; i++) {
        if (hashMap.get(nums[i]) != undefined && (i != hashMap.get(nums[i]))){
            return [i,hashMap.get(nums[i])];
        }
    }
    return [];
};